tools/clang/test/HLSLFileCheck/infra/auto-dxilver.hlsl

```diff
@@ -0,0 +1,28 @@
+// Make sure each run line either works, or implicitly requires dxilver 1.8.
+// If an implicit dxilver check is missing, we should see one of these errors.
+
+// RUN: %dxc -T vs_6_8 %s | FileCheck %s
+// This should implicitly require dxilver 1.8.
+
+// RUN: %dxc -T vs_6_8 -Vd %s | FileCheck %s
+// Even though this is using -Vd, the validator version is set by the available
+// validator.  If that isn't version 1.8 or above, we'll see an error.
+// The implicit dxilver logic should not skip the check when -Vd is used.
+// CHECK-NOT: error: validator version {{.*}} does not support target profile.
```

Comment:
Do the run lines need `2>&1` to make sure that FileCheck sees stderr for these to work?

Reply:
The point of this test is to verify a behavior that only exists in the TAEF FileCheckerTest.  It would never make sense to run this as a lit shell test.

One of the differences for FileCheckerTest is that stderr is passed to the next part automatically when the operation fails, and `2>&1` is not handled in any special way - instead, it will be passed to the compiler API and parsed as if it's a regular argument, causing it to set the `InputFile` in `hlsl::options::DxcOpts` to `"2>&1"`.

Another difference is that the lit shell test would never even get past the dxc part in any of these failure cases to even run the FileCheck in the first place.